helpers/aws_validation.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+import datetime
+import hashlib
+import hmac
+
+from urllib.error import HTTPError
+from urllib.request import Request, urlopen
+
+
+class AwsValidation:
+    """
+    A class to validate AWS credentials without using boto3 as a dependancy.
+
+    The scructure and methods have been adapted from the AWS documentation:
+    http://docs.aws.amazon.com/general/latest/gr/signature-v4-examples.html#signature-v4-examples-python
+    """
+
+    METHOD = 'POST'
+    SERVICE = 'sts'
```

Comment:
Why did you come up with `sts` and not `s3`?

Reply:
Through a bit of reading around, it seemed `sts` was a good option for general validation of AWS credentials. It also only required using the access key and secret key for validation - some other services required usernames, etc.